Source/DafnyCore/AST/AstVisitor.cs

```diff
@@ -272,6 +271,27 @@ public abstract class ASTVisitor<VisitorContext> where VisitorContext : IASTVisi
       }
     }
 
+    protected virtual void VisitExtendedPattern(ExtendedPattern pattern, VisitorContext context) {
+      switch (pattern) {
+        case DisjunctivePattern disjunctivePattern:
+          disjunctivePattern.Alternatives.ForEach(alternative => VisitExtendedPattern(alternative, context));
```

Comment:
I'm not a fan of using `ForEach` instead of `foreach`. Here's a discussion of this by Eric Lippert: https://ericlippert.com/2009/05/18/foreach-vs-foreach/, who's on the C# language designers.